src/stream/src/executor/hash_agg.rs

```diff
@@ -305,13 +343,23 @@ impl<K: HashKey, S: StateStore> HashAggExecutor<K, S> {
 
         let mut buffered = stream::iter(futures).buffer_unordered(10).fuse();
 
+        let mut afterhalf_replaced_groups = vec![];
         while let Some(result) = buffered.next().await {
             let (key, state) = result?;
-            agg_groups.put(key, Some(state));
+            Self::push_and_handle_replacement(
+                agg_groups,
+                &key,
+                Some(state),
+                &mut afterhalf_replaced_groups,
+            );
         }
         // Drop the stream manually to teach compiler the async closure above will not use the read
         // ref anymore.
         drop(buffered);
+        // Should do nothing because `afterhalf_replaced_groups` should be empty.
```

Comment:
What's this for?

Reply:
> What's this for?

to ensure that all evicted items (perhaps due to carelessness) are synced

Reply:
> What's this for?

these codes are removed now.